src/CMakeLists.txt

```diff
@@ -305,12 +305,20 @@ include_directories("${CMAKE_SOURCE_DIR}/src/dmclock/support/src")
 include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/src/googletest/googletest/include")
 
 if(WITH_SEASTAR)
-  find_package(fmt 4.0.0)
+  find_package(fmt 4.0.0 QUIET)
   if(NOT fmt_FOUND)
+    message(STATUS "Could not find fmt, will build it")
+    add_subdirectory(fmt)
+  elseif(fmt_VERSION VERSION_GREATER_EQUAL 5.0.0)
+    message(WARNING "Could NOT find fmt: "
+      "Found unsuitable version \"${fmt_VERSION}\", "
+      "but required is at most \"5.0.0\" (found ${fmt_INCLUDE_DIR}). "
+      "Will build it")
     add_subdirectory(fmt)
   endif()
-  find_package(c-ares 1.13.0 QUIET MODULE)
+  find_package(c-ares 1.13.0 QUIET)
   if(NOT c-ares_FOUND)
+    message(STATUS "Could not find c-ares, will build it")
```

Comment:
This seems to be producing output that is not used - see https://github.com/ceph/ceph/pull/22915#discussion_r200911758 where this was seen as undesirable.

Reply:
@smithfarm normally `find_packge()` shows if a packages is found, and where it is found. in this case, it does not show anything whether c-ares is found or not. for the user or developer's perspective, i am not sure it's a "good news" or not. i think they deserve to understand if a dependency is from the prebuilt package or from the source shipped in the source.

from my point of view, this message is useful. i want to know if we are using the package shipped by distro or not.

Reply:
@tchaikov Actually, I think your info messages are useful. They help when debugging various issues. That's exactly why I was perplexed when you objected to outputting the version of Cython found. IMHO this information is "useful in debugging various issues". I want to know this information. Yet simply stating this in the PR was not enough to get it past your review.

Now, you are arguing that the messages are useful and you want to see them, and I am supposed to approve the PR only on those grounds?

Do you see the disconnect here between us?

Reply:
In other words, I'm asking if you would mind if I document the following as part of our style guide: "CMake `message()` statements are OK as long as one developer thinks they are useful"

Reply:
If c-ares is not found, then the person running the build sees that c-ares submodule is being built. I could argue that the information is derivable from the presence or absence of that.

Reply:
You said (quoting): "no news is good news. just wanted to avoid unused output in success runs"

Reply:
My motivation for making a point of this is I would like to avoid arguing over such trivial issues in the future.

Reply:
You're right that the Cython version is already displayed: `-- Found Cython3: 0.27` so my message would have been redundant.

And although it's true that one can determine whether c-ares is being built from source by the presence/absence of gcc messages, these are not displayed during the cmake phase. So if you are running cmake only, then the message is not redundant.

Reply:
@smithfarm if you really want to kill this message, so be it.

Reply:
@tchaikov Can you help me understand how `Finddpdk.cmake` works in this respect? I looked at the build log and could not find any message saying "DPDK will be built from source", yet it *is* built from source.

On my py3-only dev box I am getting this message:

```
-- Could NOT find dpdk (missing: check_LIBRARIES)
```

Reply:
@tchaikov I don't want to kill the message, but I would like to understand better if there are other submodules (besides fmt and c-ares) that will not be built from source/gitmodule under certain conditions.

(Until now I was under the impression that all source code in git submodules is always built - obviously that was an oversimplification!)

Reply:
Also, I wonder whether using the system package over the gitmodule is really desirable? It seems to defeat the purpose for which the gitmodules were introduced - to ensure the same version of the code is used on all supported distros.

Reply:
> Yet it is built from source there, too.

regarding to DPDK, it's a different story. Seastar is able to use DPDK as a backend. but as we are still at an early stage of developing crimson, and will probably evaluate DPDK backend in future. that's why i haven't enabled it with DPDK.

> Until now I was under the impression that all source code in git submodules is always built 

it's not true.

> Also, I wonder whether using the system package over the gitmodule is really desirable?

we should provide it as an option for downstream and distro maintainers. submodules are introduced when the dependency cannot fulfill the requirement of Ceph, but if it can, we have no reason to compile it  from source.

> to ensure the same version of the code is used on all supported distros.

from the developer's perspective, it'd be ideal. but from the maintainer's perspective, i am afraid it's not possible.

Reply:
probably [snaps](https://en.wikipedia.org/wiki/Snappy_(package_manager)) matches your expectation. but as the upstream, i think we should always offer the options to our users.

Reply:
@tchaikov Thanks for the discussion. The system for satisfying dependencies from submodules is difficult to understand - at least for me - and the info messages that say "Did not find foo, will build it from source" help to make it a little clearer. Maybe there are other submodules for which such messages could be added?

Reply:
good idea! i understand. it's indeed confusing. because, libfmt is a new dependency here. and it's, well, quite different than others, in the sense that only `4.0.0 <= FMT_VERSION < 5.0.0` can satisfy the requirement of seastar, and also not all distro offer `libfmt-{dev,devel}`. all these render  that we cannot have libfmt-dev or libfmt-devel in rpm.spec or d/control. and i just learned this yesterday with your help, as my testbed happens to have libfmt-dev 4.1.0.

will prepare a follow-up PR to address this to clear up the confusion.

Reply:
As a personal note, I'd rather have the CMake output to be TO verbose. 
The make output is already large enough that a regular 'user' cannot easily find any information is this haystack.

Since if there is information missing, the only option as a developer is to locally add it to figure out the problem and start redoing cmake.
And there is no better knowledge than the original developer to add useful diagnostics.

So I'd prefer not to be too pedantic about adding diagnostics. 
But then again, Kefu is the CMake ninja here, so I do respect his view on things.